test/extended/util/metrics.go

```diff
@@ -0,0 +1,352 @@
+			info.result = fmt.Sprintf("error: failed to find counter: %q", "rest_client_requests_total")
+			info.failed = true
+			podInfos = append(podInfos, info)
+			continue
+		}
+
+		procStartTime, ok := metrics["process_start_time_seconds"]
+		if !ok || len(procStartTime.Metric) == 0 {
+			info.result = fmt.Sprintf("error: failed to find metric: %q", "process_start_time_seconds")
+			info.failed = true
+			podInfos = append(podInfos, info)
+			continue
+		}
+		procStartTimeSeconds := procStartTime.Metric[0].GetGauge().GetValue()
+		totalProcTimeSeconds := time.Now().Unix() - int64(procStartTimeSeconds)
+
+		totalRequestCount := float64(0)
+		for _, metric := range metricGroup.Metric {
+			totalRequestCount += metric.Counter.GetValue()
+		}
+
+		comment := "within QPS bounds"
+		qps := totalRequestCount / float64(totalProcTimeSeconds)
+		if qps > maxQPSAllowed {
+			comment = "exceeds QPS bounds"
+		}
+		info.status = fmt.Sprintf("%s (%s)", info.status, comment)
+		info.qps = qps
+		info.result = fmt.Sprintf("%v requests over a span of %v seconds", totalRequestCount, totalProcTimeSeconds)
+		podInfos = append(podInfos, info)
+	}
+
+	return podInfos, nil
+}
+
+func getBearerToken(adminClient kubernetes.Interface, namespace string) (string, error) {
+	if len(calculatedBearerToken) > 0 {
+		return calculatedBearerToken, nil
+	}
+
+	sa := &v1.ServiceAccount{
+		TypeMeta:   metav1.TypeMeta{APIVersion: v1.SchemeGroupVersion.String(), Kind: "ServiceAccount"},
+		ObjectMeta: metav1.ObjectMeta{Name: namespace + "-sa", Namespace: namespace},
+	}
+
+	// ensure our namespace exists
+	_, err := adminClient.CoreV1().Namespaces().Create(&v1.Namespace{
+		TypeMeta:   metav1.TypeMeta{APIVersion: v1.SchemeGroupVersion.String(), Kind: "Namespace"},
+		ObjectMeta: metav1.ObjectMeta{Name: namespace},
+	})
+	if err != nil && !kapierrs.IsAlreadyExists(err) {
+		return "", err
+	}
+
+	// ensure our service account can GET /metrics
+	_, err = adminClient.RbacV1().ClusterRoleBindings().Create(&rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{APIVersion: rbacv1.SchemeGroupVersion.String(), Kind: "ClusterRoleBinding"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace + "-metrics-viewer",
+		},
+		RoleRef: rbacv1.RoleRef{APIGroup: rbacv1.GroupName, Kind: "ClusterRole", Name: "cluster-admin"},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      sa.Name,
+				Namespace: namespace,
+			},
+		},
+	})
+	if err != nil && !kapierrs.IsAlreadyExists(err) {
+		return "", err
+	}
+
+	_, err = adminClient.CoreV1().ServiceAccounts(namespace).Create(sa)
+	if err != nil && !kapierrs.IsAlreadyExists(err) {
+		return "", err
+	}
+
+	// create secret to store the generated token for our serviceaccount
+	saSecret, err := adminClient.CoreV1().Secrets(namespace).Create(&v1.Secret{
+		TypeMeta: metav1.TypeMeta{APIVersion: v1.SchemeGroupVersion.String(), Kind: "Secret"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      namespace + "-sa-token",
```

Comment:
nit:I am not a fan of naming objects by prefixing them with the namespace as that's why you have the namespace in the first place :)